### PR TITLE
Should you commit your lock file blog

### PR DESCRIPTION
**Pull request by akshayvadher**
## What did you do?

Please include a summary of the changes.

- [x] Added a blog on whether to commit lock file or not

## Why did you do it?

Even after working in with the nodejs for long time, many people still don't know whether to commit lock file or not

### CODE REVIEW

**Review by 2KAbhishek — APPROVED** [review]
Looks good, made tiny fix

**Comment by akshayvadher** [comment]
> Looks good, made tiny fix

I ran it through Grammarly, not sure why it was missed. Thanks.

File: content/blog/commit-lock-file.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+a package `foo` is defined to use `^6.2.8`, and you run `npm install`, it will install the latest version of `6.x.x`
+after `6.2.8`.
+
+# Issues
+
+## Deterministic builds
+
+It is a nicer way to say, "It works on my machine".
+
+Let's say I use `foo` with version `^6.2.8` and use `npn install`, and it will install the latest `6.2.9`. I commit my
```
Comment (utsav00) [inline content/blog/commit-lock-file.md:32]: I think you meant `npm install` here

**Comment by akshayvadher** [comment]
> @akshayvadher Should we talk about putting the exact version of packages in package.json instead of `~` and `^` and how that still isn't an alternative to committing lock file?

Good catch. I thought I wrote it!

File: content/blog/commit-lock-file.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+image = "/images/2023/commit-lock-file/version-locking-case.png"
+draft = false
+authors = ["Akshay Vadher"]
+description = ""
+tags = ["Guideline", "Best Practice", "Build", "Tooling", "Lock File", "Package Manager", "npm", "yarn", "pnpm", "bun"]
+categories = ["Guideline", "Best Practice", "Software Craftsmanship"]
+type = ""
++++
+
+# _tl;dr;_
```
Comment (utsav00) [inline content/blog/commit-lock-file.md:14]: Nit: no need for semicolon at the end. Should be `TL;DR`

**Review by utsav00 — APPROVED** [review]

File: content/blog/commit-lock-file.md (hunk 2 above)
Comment (akshayvadher) [reply content/blog/commit-lock-file.md:14]: image
or
image
or
image
---
If we consider it a header, the header's first character is always the title case.

File: content/authors/akshay-vadher/_index.md [hunk 3]
```diff
@@ -5,7 +5,10 @@ image: '/authors/akshay-vadher/akshay-vadher.jpg'
 ---
 <img src="akshay-vadher.jpg" alt="Akshay Vadher profile" width="150"/>
 
-
 Getting things done.
 
-I vent my thoughts [here](https://akshayvadher.github.io/).
\ No newline at end of file
+[![GitHub](../../icons/github.svg)](https://github.com/akshayvadher)
```
Comment (2KAbhishek) [inline content/authors/akshay-vadher/_index.md:10]: nit: these svgs don't have good visibility on dark theme, we have font awesome css on the blog, using icons from there will use the correct colors

**Pull request merged**